### PR TITLE
Adding cost733class

### PR DESCRIPTION
**Pull request by cofinoa**
Checklist
- [X] Title of this PR is meaningful: e.g. "Adding my_nifty_package", not "updated meta.yaml".
- [X] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/5eddbd7fc9d1502169089da06c3688d9759be978/recipes/example/meta.yaml#L64-L73) for an example).
- [X] Source is from official source.
- [X] Package does not vendor other packages. (If a package uses the source of another package, they should be separate packages or the licenses of all packages need to be packaged).
- [X] If static libraries are linked in, the license of the static library is packaged.
- [X] Package does not ship static libraries. If static libraries are needed, [follow CFEP-18](https://github.com/conda-forge/cfep/blob/main/cfep-18.md).
- [X] Build number is 0.
- [X] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details).
- [X] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there.
- [X] When in trouble, please check our [knowledge base documentation](https://conda-forge.org/docs/maintainer/knowledge_base.html) before pinging a team.

### CODE REVIEW

**Comment by cofinoa** [comment]
Dear @conda-forge/staged-recipes , 

please, any volunteer to review?

**Comment by cofinoa** [comment]
This a recipe for a FORTRAN software, but I haven't received any feedback to review it from @conda-forge/staged-recipes 

Please, someone from @conda-forge/help-c-cpp could review?

**Comment by ocefpaf** [comment]
@cofinoa I had a few comments on a "pending review" that I completely forgot. Take a look at them so we can sort this recipe out.

File: recipes/cost733class/bld.bat [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+  -D CMAKE_BUILD_TYPE=%BUILD_TYPE% ^
+  -D GRIB=OFF ^
+  -D NCDF=ON ^
+  -D OPENGL=OFF ^
+  %SRC_DIR%
+if errorlevel 1 exit 1
+
+cmake --build . --target install --config %BUILD_TYPE%
+::make
+::make install
```
[the remark this answers is not in the document]
Comment (cofinoa) [reply recipes/cost733class/bld.bat:32]: I have removed the extra "make" lines

No, there is no check/test. 

Just the test in the recipe checking that the binary has been created

File: recipes/cost733class/build.sh [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+  -D CMAKE_BUILD_TYPE=${BUILD_TYPE} \
+  -D GRIB=${ENABLE_GRIB} \
+  -D NCDF=ON \
+  -D OPENGL=OFF \
+  $SRC_DIR
+
+
+#cmake --build . --target install --config Release
+make 
+make install
```
[the remark this answers is not in the document]
Comment (cofinoa) [reply recipes/cost733class/build.sh]: I have removed the extra "make" lines

No, there is no check/test. 

Just the test in the recipe checking that the binary has been created

**Comment by cofinoa** [comment]
@ocefpaf thank you for taking care of this. 

I have removed the extra "make" lines in both build scripts and left just "cmake" commands.

I have resolved your reviews.

File: recipes/cost733class/meta.yaml [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+
+build:
+  skip: True  # [win]
+  number: 0
+
+requirements:
+  build:
+    - cmake
+
+    # Unix
```
Comment (ocefpaf) [inline recipes/cost733class/meta.yaml:20]: With the pre-processor selectors this comment and the one below aren't really needed, right?

```suggestion
```

File: recipes/cost733class/meta.yaml [hunk 4]
```diff
@@ -0,0 +1,63 @@
...
+    - {{ compiler('c') }}              # [unix]
+    - {{ compiler('cxx') }}            # [unix]
+    - {{ compiler('fortran') }}        # [unix]
+    - make                             # [unix]
+    - pkg-config                       # [unix]
+    
+    # Win
```
Comment (ocefpaf) [inline recipes/cost733class/meta.yaml:27]:
```suggestion

```

File: recipes/cost733class/meta.yaml [hunk 5]
```diff
@@ -0,0 +1,63 @@
...
+  home: https://projects.met.no/cost733/
+  license: GPL-3.0-only
+  license_family: GPL
+  license_file: LICENSE
+  summary: 'Program package for weather and circulation type classification.'
+  description: |
+    cost733class is a program package for weather and circulation type classification.
+    Copyright (C) 2013 Andreas Philipp
+  doc_url: https://git.rz.uni-augsburg.de/philipan/cost733class-1.4/-/raw/main/doc/cost733class_userguide.pdf
+  dev_url: https://github.com/lochbika/{{ name }}
```
Comment (ocefpaf) [inline recipes/cost733class/meta.yaml:59]: Let's not abuse the jinja variable. It is nice to make this URL clickable with the proper name in it.

```suggestion
  dev_url: https://github.com/lochbika/cost733class
```

File: recipes/cost733class/bld.bat [hunk 6]
```diff
@@ -0,0 +1,31 @@
...
+  -D CMAKE_INSTALL_PREFIX=%LIBRARY_PREFIX% ^
+  -D CMAKE_PREFIX_PATH="%LIBRARY_PREFIX%" ^
+  -D CMAKE_BUILD_TYPE=%BUILD_TYPE% ^
+  -D GRIB=OFF ^
+  -D NCDF=ON ^
+  -D OPENGL=OFF ^
+  %SRC_DIR%
+if errorlevel 1 exit 1
+
+cmake --build . --target install --config %BUILD_TYPE%
```
Comment (ocefpaf) [inline recipes/cost733class/bld.bat:30]: Is there a `make check` or `ctest` we can run? I'm not sure if I asked this already but I cannot find a comment on this.

Comment (cofinoa) [reply recipes/cost733class/bld.bat:30]: No, there is no check/ctest.

Just there is a test in the recipe checking that the binary has been created

**Comment by cofinoa** [comment]
@ocefpaf , I have accepted your suggestions. 

With respect to the make check/ctest, there is no test in the original source. I have just added a test in the recipe, to check the executable it's been generated. 

Thanks!

File: recipes/cost733class/bld.bat (hunk 6 above)
Comment (ocefpaf) [reply recipes/cost733class/bld.bat:30]: > Just there is a test in the recipe checking that the binary has been created

I wonder if there is anything we can to test it better than just that check?

Comment (cofinoa) [reply recipes/cost733class/bld.bat:30]: The original doc:
https://git.rz.uni-augsburg.de/philipan/cost733class-1.4/-/blob/main/doc/cost733class_userguide.pdf

has some examples with an sample data file (`slp.dat`) in the source repo, using different methods of classifications, but they are too large in CPU, WALLTIME and MEMORY, for a test. 

The original source repo, also has some example scripts, but for GUI output which is not suitable for test the build. 

If you don't mind, I would suggest just to left the current check and sort the recipe out.

Comment (ocefpaf) [reply recipes/cost733class/bld.bat:30]: > but they are too large in CPU, WALLTIME and MEMORY, for a test.

How large? Do you mind testing it out first and, if it fails here due to lack of resources, we can leave the simpler test.

**Pull request merged**